TELCODOCS-261-content: Branch to add all Redfish topics (#41289)
openshift/openshift-docs: modules/baremetal-hardware-event-proxy-operator.adoc

// Module included in the following assemblies:
//
// * operators/operator-reference.adoc
[id="baremetal-hardware-event-proxy_{context}"]
= Baremetal Hardware Event Proxy Operator

[discrete]
== Purpose

> The OpenShift Baremetal Hardware Event Proxy Operator enables applications running on bare-metal clusters to respond quickly to Redfish hardware changes and failures such as breaches of temperature thresholds, fan failure, disk loss, power outages, and memory failure. The operator can be viewed with:

Reviewer: The OpenShift Baremetal Hardware Event Proxy Operator _manages life-cycle of Baremetal Hardware Event proxy_ , which enables applications running on bare-metal clusters to respond.....

Author: Done.

[source,terminal]
----

> $ oc -n [namespace] get cm hw-event-proxy-operator-manager-config -o yaml

Reviewer: no need for this

Author: Removed.

----

> The custom resource definition (CRD) `HardwareEvent` Kind can be viewed in a cluster with:

Author: Done.

[source,terminal]
----
$ oc get crd hardwareevents.event.redhat-cne.org -o yaml
----

> you can edit configuration objects with:

Reviewer: Configuration objects

oc -n [namespace] edit cm hw-event-proxy-operator-manager-config

// Module included in the following assemblies:
//
// * operators/operator-reference.adoc
[id="baremetal-hardware-event-proxy_{context}"]
= Baremetal Hardware Event Proxy Operator

[discrete]
== Purpose
The OpenShift Baremetal Hardware Event Proxy Operator manages the life-cycle of the Baremetal Hardware Event proxy. The proxy enables applications running on bare-metal clusters to respond quickly to Redfish hardware changes and failures such as breaches of temperature thresholds, fan failure, disk loss, power outages, and memory failure.

== Configuration Objects
You can edit configuration objects with:
[source,terminal]
----

> $ oc -n [namespace] edit cm hw-event-proxy-operator-manager-config

Author: The customer chooses the namespace at installation. We cannot specify. Stet.

> $ oc -n [namespace] edit cm hw-event-proxy-operator-manager-config

Author: Added detail to describe the most typical use of the command.

----
[discrete]

== Project

> link:https://github.com/redhat-cne/hw-event-proxy-operator[hw-event-proxy-operator]

Author: All the other Operator topics have just a link to the project. We are leaving it in line with the other topics; Stet.

== CRDs
`hardwareevents.event.redhat-cne.org`:

* Scope: Namespaced
* CR: HardwareEvent

> * Validation: Yes

Author: Moved a description  of the CR purpose to this section.

// Module included in the following assemblies:
//
// * operators/operator-reference.adoc
[id="baremetal-hardware-event-proxy_{context}"]
= Baremetal Hardware Event Proxy Operator

[discrete]
== Purpose

> The OpenShift Baremetal Hardware Event Proxy Operator manages the life-cycle of the Baremetal Hardware Event proxy. The Baremetal Hardware Event Proxy enables the user to configure the types of event that are monitored using Redfish Hardware Events on the cluster.

Author: The name is widely established. Steve Smith on ZTP has the same issue with his developers.
Fixed back ticks.

== Configuration Objects
You can use this command to edit the configuration after installation: for example, the webhook port.
You can edit configuration objects with:
[source,terminal]
----
$ oc -n [namespace] edit cm hw-event-proxy-operator-manager-config
----
[source,terminal]
----
apiVersion: controller-runtime.sigs.k8s.io/v1alpha1
kind: ControllerManagerConfig
health:
  healthProbeBindAddress: :8081
metrics:
  bindAddress: 127.0.0.1:8080
webhook:
  port: 9443
leaderElection:
  leaderElect: true
  resourceName: 6e7a703c.redhat-cne.org
----
[discrete]

== Project
link:https://github.com/redhat-cne/hw-event-proxy-operator[hw-event-proxy-operator] 

== CRD

> The proxy enables applications running on bare-metal clusters to respond quickly to Redfish hardware changes and failures such as breaches of temperature thresholds, fan failure, disk loss, power outages, and memory failure, reported using the HardwareEvent CR.

Reviewer: Suggest move this description under `hardwareevents.event.redhat-cne.org`:
```suggestion
This defines defines the `Baremetal Hardware Event proxy`, which enables applications running on bare-metal clusters to respond quickly to Redfish hardware changes and failures such as breaches of temperature thresholds, fan failure, disk loss, power outages, and memory failure.
```

Author: Done.
